crosschain-resolver/contracts/L1Resolver.sol

```diff
@@ -177,38 +203,100 @@ contract L1Resolver is EVMFetchTarget {
         string memory key,
         address target
     ) private view returns (bytes memory) {
-        EVMFetcher.newFetchRequest(verifier, target)
+        EVMFetcher
+            .newFetchRequest(verifier, target)
             .getStatic(RECORD_VERSIONS_SLOT)
-              .element(node)
+            .element(node)
             .getDynamic(VERSIONABLE_TEXTS_SLOT)
-              .ref(0)
-              .element(node)
-              .element(key)
-            .fetch(this.textCallback.selector, '');
+            .ref(0)
+            .element(node)
+            .element(key)
+            .fetch(this.textCallback.selector, msg.data[0:4]);
     }
 
     function textCallback(
         bytes[] memory values,
-        bytes memory
+        bytes memory sig
     ) public pure returns (bytes memory) {
+        if (keccak256(sig) != keccak256(hex"9061b923")) {
+            return bytes(string(values[1]));
+        }
         return abi.encode(string(values[1]));
     }
 
-    function _contenthash(bytes32 node, address target) private view returns (bytes memory) {
-        EVMFetcher.newFetchRequest(verifier, target)
+    function _contenthash(
+        bytes32 node,
+        address target
+    ) private view returns (bytes memory) {
+        EVMFetcher
+            .newFetchRequest(verifier, target)
             .getStatic(RECORD_VERSIONS_SLOT)
-              .element(node)
+            .element(node)
             .getDynamic(VERSIONABLE_HASHES_SLOT)
-              .ref(0)
-              .element(node)
-            .fetch(this.contenthashCallback.selector, '');
+            .ref(0)
+            .element(node)
+            .fetch(this.contenthashCallback.selector, msg.data[0:4]);
     }
 
     function contenthashCallback(
         bytes[] memory values,
-        bytes memory
+        bytes memory sig
     ) public pure returns (bytes memory) {
+        if (keccak256(sig) != keccak256(hex"9061b923")) {
+            return values[1];
+        }
         return abi.encode(values[1]);
     }
 
+    /**
+     * Returns the address associated with an ENS node.
+     * @param node The ENS node to query.
+     * @return The associated address.
+     */
+    function addr(
```

Comment:
As per the other PR - we shouldn't be implementing these legacy methods. `resolve` is intended to replace them entirely.

Reply:
I have reverted the changes